lib/FieldType/DataTransformer/RichTextValueTransformer.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+namespace EzSystems\RepositoryForms\FieldType\DataTransformer;
+
+use eZ\Publish\API\Repository\FieldType;
+use eZ\Publish\Core\FieldType\RichText\Value;
+use Symfony\Component\Form\DataTransformerInterface;
+
+/**
+ * DataTransformer for RichText\Value.
+ */
+class RichTextValueTransformer implements DataTransformerInterface
+{
+    /** @var FieldType */
+    private $fieldType;
+
+    public function __construct(FieldType $fieldType)
+    {
+        $this->fieldType = $fieldType;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return string
+     */
+    public function transform($value)
+    {
+        if (!$value instanceof Value) {
+            return '';
+        }
+
+        return (string) $value;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return Value|null
+     */
+    public function reverseTransform($value)
+    {
+        if ($value === null || empty($value)) {
+            return $this->fieldType->getEmptyValue();
+        }
+
+        return $this->fieldType->fromHash(['xml' => $value]);
```

Comment:
and maybe same here, is it enough to accept an `xhtml5edit` document ?

Reply:
Funny fact - This is how it works in PlatformUI. UI is sending JSON object with `xml` and `xhtml5edit` properties but both contain the same markup (xhtml5edit). Kinda hackish but it is later picked up by a processor which is identifying the schema and converting to docbook. It works the way its implemented above as `fromHash()` is taking care of it.